lib/llvmopencl/TargetAddressSpaces.cc

```diff
@@ -92,6 +92,9 @@ TargetAddressSpaces::runOnModule(llvm::Module &M) {
 
   llvm::StringRef arch(M.getTargetTriple());
 
+  if (arch.startswith("spir"))
+    arch = OCL_KERNEL_TARGET;
```

Comment:
This works only for host devices? Remember that pocl can control platforms with multiple devices that are not the same as host. And the same SPIR can be retargeted to all devices in the platform (which could consist of, say a MIPS CPU and a HSA-supported GPU in the future).   So a better fix would be to dig out the device we are compiling to somehow.